src/index.js

```diff
@@ -1,5 +1,7 @@
 const factorial = (number) => {
-    
-  }
+  let result = 1, i = number;
```

Comment:
quita ` i = number` dentro del for se crea otra variable i con scope únicamente dentro del for y no se afecta esta i. Por lo que no se usa.

Reply:
ahora lo hice ahora de forma recursiva hector antes de ver tu publicación :D